doc-site/docs/architecture/distributed_sequencer.md

```diff
@@ -0,0 +1,1437 @@
+  else
+    R --) S: Error
+  end
+```
+
+### Message specification and handler responsibilities
+
+The following concrete message exchanges defines the responsibilities and expectation of each node in the network.
+
+
+
+#### <a name="message-transaction-delegation-command"></a>Transaction delegation
+
+
+This is an instance of the `Command` pattern where the `command sender` is the sender node for the given transaction and the `command receiver` is the node that has been chosen, by the `sender` as the `coordinator`.  The `sender` sends a `DelegationCommand` message...
+
+```mermaid
+sequenceDiagram
+  %%{init: { "sequence": { "mirrorActors":false }}}%%
+  participant S as Sender
+  participant C as Coordinator
+  loop Every RetryInterval
+    S -) C: DelegationCommand
+  end
+  alt delegation valid
+    C --) S: DelegationAccepted
+    C -->> C: Coordinate Transaction
+    activate C
+    deactivate C
+  else
+    C --) S: DelegationRejected
+  end
+  
+```
+
+
+##### <a name="message-transaction-delegation-command"></a>Delegation Command
+```proto
+message DelegationCommand {
+    string transaction_id = 1;
+    string delegate_node_id = 2;
+    string delegation_id = 3; //this is used to correlate the acknowledgement back to the delegation. unlike the transport message id / correlation id, this is not unique across retries
+    bytes private_transaction = 4; //json serialized copy of the in-memory private transaction object
+    int64 block_height = 5; // the block height upon which this delegation was calculated (the highest delegation wins when crossing in the post)
+}
+```
+
+As a member of the coordinator committee, a node that receives a `DelegationCommand` must either respond by sending `DelegationAccepted` or `DelegationRejected` message.  
+
+Valid reasons for rejection are
+ - requesters block height is in a different block range than expected.  In this case, the `DelegationRejected` message must contain the block height that the local node is aware of so that the delegating node can make an informed decision based on whether they are behind or ahead.
+ - the current node is not the most preferred available coordinator for the current block height.
+
+If the Delegation is accepted, then the receiving node must begin [coordinating that transaction](#subprocess-coordinate-transaction)
+
+```mermaid
+---
+title: OnDelegationCommand
+---
+flowchart TB
+    A@{ shape: sm-circ, label: "Start" }
+    B@{ shape: diam, label: "Check <br/>block<br/>range" }
+    C@{ shape: lean-r, label: "Reject<br/>(wrong block height)" }
+    D@{ shape: fr-rect, label: "Select<br/>available<br/>coordinator" }
+    E@{ shape: diam, label: "this Is<br/>Coordinator" }
+    I@{ shape: lean-r, label: "Reject<br/>(wrong coordinator)" }
+    G@{ shape: fr-rect, label: "<a href="#subprocess-coordinate-transaction">coordinate transaction</a>" }
+    J@{ shape: fr-circ, label: " " }
+
+
+    A-->B
+    B --> |Behind| C
+    B --> |Ahead| C
+    B --> |Same| D
```

Comment:
I don't understand yet how this works. I understand the "I've not indexed that block yet" problem, but this algorithm seems to require an **exact** block match, and that mean a lot of thrashing on block boundaries. For example our dev environments come pre-configured with a 100ms (non-empty) block time, so the likelihood under even minimal  load of all nodes being on an identical block is low. With a more traditional 2sec block time, there's definitely a slightly wider window, but I still don't think we should optimize for exact block matching the whole time.

Isn't the _range_ more important?

e.g. the three paths are:
- Agreement: I've indexed up to that block, and I agree on your assessment of who the coordinator is
- Behind: I've not indexed that block yet
- Too far ahead: I've indexed so far ahead, that I think we're in the next range and I'm not the coordinator anymore.